EasyReflectometry/sample/sample.py

```diff
@@ -91,3 +71,24 @@ def as_dict(self, skip: list = None) -> dict:
         for i, layer in enumerate(self.data):
             this_dict['data'][i] = layer.as_dict()
         return this_dict
+
+    @classmethod
+    def from_dict(cls, data: dict) -> Sample:
+        """
+        Create a Sample from a dictionary.
+
+        :param data: dictionary of the Sample
+        :return: Sample
+        """
+        sample = super().from_dict(data)
+
+        # Remove the default multilayers
+        for i in range(NR_DEFAULT_LAYERS):
+            sample.__delitem__(0)
+
+        # Ensure that the data is also converted
+        # TODO Should probably be handled in EasyCore
+        for i in range(len(sample.data)):
+            sample[i] = sample[i].__class__.from_dict(data['data'][i])
```

Comment:
the only type `sample[i]` can have is `Sample` I think (correct me if I'm wrong!)
If so, wouldn't it be easier to write
```python
 sample[i] = Sample.from_dict(data['data'][i])
```

Reply:
We are actually looping over the assemblies that make up the sample